src/StateMachine.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * This file is part of the php-state project.
+ *
+ * (c) Yannick Voyer <star.yvoyer@gmail.com> (http://github.com/yvoyer)
+ */
+
+namespace Star\Component\State;
+
+use Star\Component\State\Attribute\StringAttribute;
+use Star\Component\State\Event\ContextTransitionWasRequested;
+use Star\Component\State\Event\ContextTransitionWasSuccessful;
+use Star\Component\State\Event\StateEventStore;
+use Star\Component\State\Event\TransitionWasSuccessful;
+use Star\Component\State\Event\TransitionWasRequested;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+
+final class StateMachine
+{
+    /**
+     * @var EventDispatcher
+     */
+    private $dispatcher;
+
+    /**
+     * @var FailureHandler
+     */
+    private $failureHandler;
+
+    /**
+     * @var TransitionRegistry
+     */
+    private $registry;
+
+    public function __construct()
+    {
+        $this->dispatcher = new EventDispatcher();
+        $this->failureHandler = new AlwaysThrowException();
+//        $this->transitionFactory = new TransitionBuilder();
+        $this->registry = new TransitionRegistry();
+    }
+
+    /**
+     * @param string $name
+     * @param StateContext $context
+     */
+    public function transitContext($name, StateContext $context)
+    {
+        $alias = $context->contextAlias();
+        $this->registry->useFailureHandler($this->failureHandler);
+        $transition = $this->registry->getTransition($name, $alias);
```

Comment:
Registry should receive as arg the handler. Remove handler from registry attribute